### PR TITLE
Upgrades to Chipmunk v7.0

### PR DESCRIPTION
**Pull request by ricardoquesada**
( This PR is based on this one with some fixes and more tests: https://github.com/cocos2d/cocos2d-x/pull/14949 )
- Upgrades to Chipmunk v7.0
  - multithreaded
  - faster (Neon optimized + 2 threads)
- Removes `PhysicsDebugDraw` (uses internally `DrawNode`)
- Updates JS bindings
- Chipmunk JS bindings are generated **AUTOMATICALLY** (do not manually edit them)
- The JS API is about 95% backwards compatible... there a few things that are not backwards compatible due to the changes in the API in Chipmunk
- Build systems updated

### CODE REVIEW

**Comment by TheCodez** [comment]
@ricardoqueasada cpHastySpace doesnt work in Windows as it uses pthreads

**Comment by ricardoquesada** [comment]
@TheCodez Ok. Thanks. I'll disable `cpHastySpace` for windows

**Comment by TheCodez** [comment]
There is a lib that adds pthread on windows. https://www.sourceware.org/pthreads-win32/

**Comment by stammen** [comment]
@ricardoquesada The Windows 8.1 and 10.0 Universal apps have a pthread shim in https://github.com/MSOpenTech/cocos2d-x/blob/v3/cocos/platform/winrt/CCPThreadWinRT.h

I might need to add a few more pthread functions to support cpHastySpace.

**Comment by TheCodez** [comment]
@stammen those wrapper functions should work on win32, too. Why not add them?

File: cocos/physics/CCPhysicsWorld.cpp [hunk 1]
```diff
@@ -96,58 +95,60 @@ class PhysicsWorldCallback
...
     return world->collisionBeginCallback(*contact);
+    return true;
 }
 
-int PhysicsWorldCallback::collisionPreSolveCallbackFunc(cpArbiter *arb, cpSpace *space, PhysicsWorld *world)
+cpBool PhysicsWorldCallback::collisionPreSolveCallbackFunc(cpArbiter *arb, cpSpace *space, PhysicsWorld *world)
 {
-    return world->collisionPreSolveCallback(*static_cast<PhysicsContact*>(arb->data));
+    return world->collisionPreSolveCallback(*static_cast<PhysicsContact*>(cpArbiterGetUserData(arb)));
+    return true;
```
Comment (TheCodez) [inline cocos/physics/CCPhysicsWorld.cpp:117]: wrong second return.

**Comment by ricardoquesada** [comment]
deprecated. this PR is the new one:
https://github.com/cocos2d/cocos2d-x/pull/15138

File: cocos/physics/CCPhysicsWorld.cpp (hunk 1 above)
Comment (ricardoquesada) [inline cocos/physics/CCPhysicsWorld.cpp:117]: Thanks!

**Pull request closed without merging**